src/Core/OrganizationFeatures/OrganizationUsers/AcceptOrgUserCommand.cs

```diff
@@ -0,0 +1,224 @@
+    {
+        var org = await _organizationRepository.GetByIdentifierAsync(orgSsoIdentifier);
+        if (org == null)
+        {
+            throw new BadRequestException("Organization invalid.");
+        }
+
+        var orgUser = await _organizationUserRepository.GetByOrganizationAsync(org.Id, user.Id);
+        if (orgUser == null)
+        {
+            throw new BadRequestException("User not found within organization.");
+        }
+
+        return await AcceptOrgUserAsync(orgUser, user, userService);
+    }
+
+    public async Task<OrganizationUser> AcceptOrgUserByOrgIdAsync(Guid organizationId, User user, IUserService userService)
+    {
+        var org = await _organizationRepository.GetByIdAsync(organizationId);
+        if (org == null)
+        {
+            throw new BadRequestException("Organization invalid.");
+        }
+
+        var orgUser = await _organizationUserRepository.GetByOrganizationAsync(org.Id, user.Id);
+        if (orgUser == null)
+        {
+            throw new BadRequestException("User not found within organization.");
+        }
+
+        return await AcceptOrgUserAsync(orgUser, user, userService);
+    }
+
+    public async Task<OrganizationUser> AcceptOrgUserAsync(OrganizationUser orgUser, User user,
+        IUserService userService)
+    {
+        if (orgUser.Status == OrganizationUserStatusType.Revoked)
+        {
+            throw new BadRequestException("Your organization access has been revoked.");
+        }
+
+        if (orgUser.Status != OrganizationUserStatusType.Invited)
+        {
+            throw new BadRequestException("Already accepted.");
+        }
+
+        if (orgUser.Type == OrganizationUserType.Owner || orgUser.Type == OrganizationUserType.Admin)
+        {
+            var org = await _organizationRepository.GetByIdAsync(orgUser.OrganizationId);
+            if (org.PlanType == PlanType.Free)
+            {
+                var adminCount = await _organizationUserRepository.GetCountByFreeOrganizationAdminUserAsync(
+                    user.Id);
+                if (adminCount > 0)
+                {
+                    throw new BadRequestException("You can only be an admin of one free organization.");
+                }
+            }
+        }
+
+        // Enforce Single Organization Policy of organization user is trying to join
+        var allOrgUsers = await _organizationUserRepository.GetManyByUserAsync(user.Id);
+        var hasOtherOrgs = allOrgUsers.Any(ou => ou.OrganizationId != orgUser.OrganizationId);
+        var invitedSingleOrgPolicies = await _policyService.GetPoliciesApplicableToUserAsync(user.Id,
+            PolicyType.SingleOrg, OrganizationUserStatusType.Invited);
+
+        if (hasOtherOrgs && invitedSingleOrgPolicies.Any(p => p.OrganizationId == orgUser.OrganizationId))
+        {
+            throw new BadRequestException("You may not join this organization until you leave or remove " +
+                "all other organizations.");
```

Comment:
question: Why are we splitting/joining this exception message? (And some others below) 

It looks like it may be due to line length, but we don't do it consistently elsewhere?

(I see this is mostly copy pasta from the org service so not a blocking concern, just something odd I noticed)

Reply:
I removed all the joins to make things consistent.